src/core/syscalls/syscall_handler.rs

```diff
@@ -1 +1,61 @@
-pub trait SyscallHandler {}
+use cairo_rs::{types::relocatable::Relocatable, vm::vm_core::VirtualMachine};
+use felt::Felt252;
+use num_traits::Zero;
+
+use crate::core::errors::{state_errors::StateError, syscall_handler_errors::SyscallHandlerError};
+
+use super::{
+    syscall_request::{FromPtr, StorageReadRequest, SyscallRequest},
+    syscall_response::{ResponseBody, SyscallResponse},
+};
+
+pub(crate) trait SyscallHandler {
+    fn storage_read(
+        &mut self,
+        vm: &VirtualMachine,
+        syscall_ptr: Relocatable,
```

Comment:
the function should receive the request instead of the pointer because in the new version of cairo lang, this call is invoked by `syscall()` that already reads the request from memory.